ports/stm32/adc.c

```diff
@@ -249,11 +292,25 @@ STATIC void adc_config_channel(ADC_HandleTypeDef *adc_handle, uint32_t channel)
     sConfig.SamplingTime = ADC_SAMPLETIME_12CYCLES_5;
     sConfig.SingleDiff = ADC_SINGLE_ENDED;
     sConfig.OffsetNumber = ADC_OFFSET_NONE;
+#elif defined(STM32H7)
+    sConfig.SamplingTime = ADC_SAMPLETIME_8CYCLES_5;
+    sConfig.SingleDiff   = ADC_SINGLE_ENDED;
+    sConfig.OffsetNumber = ADC_OFFSET_NONE;
+    sConfig.OffsetRightShift       = DISABLE;
+    sConfig.OffsetSignedSaturation = DISABLE;
 #else
     #error Unsupported processor
 #endif
     sConfig.Offset = 0;
 
+    if (ADC_IS_INTERNAL_CHANNEL(channel)) {
+#if defined(STM32H7)
+        sConfig.SamplingTime = ADC_SAMPLETIME_810CYCLES_5;
+#else
+        sConfig.SamplingTime = ADC_SAMPLETIME_480CYCLES;
+#endif
```

Comment:
Is this a change in functionality, that the sample time is increased for internal channels?

Reply:
There's a minimum 9-10 us sampling time for internal channels, I thought I should use the highest value to be on the safe side. will revert this.